src/offlinelogstorage/dlt_offline_logstorage_behavior.c

```diff
@@ -382,11 +382,13 @@ int dlt_logstorage_storage_dir_info(DltLogStorageUserConfig *file_config,
         if (strncmp(files[i]->d_name, file_name, len) == 0) {
             if (config->num_files == 1 && file_config->logfile_optional_counter) {
                 /* <filename>.dlt or <filename>_<tmsp>.dlt */
-                if ((files[i]->d_name[len] == suffix[0]) ||
+                if (strcmp(files[i]->d_name + len, suffix) == 0 ||
```

Comment:
Do you mean it is wrong also as it is? Because all I did was change the suffix comparison from only comparing the "." to comparing the entire suffix.

If we need to change the comparison for files with timestamps then maybe that would be a different pull request?

Reply:
@averater 
I know, it should be covered all cases in your PR.
You can apply in current PR, don't need other PR

Reply:
I would prefer if someone else fixed that part as I have other task with higher prio and don't have time to look into this soon. Either merge only this PR and fix only this part or someone else has to fix all of it.